front/app/utils/streams/index.ts

```diff
@@ -995,7 +879,106 @@ class Streams {
 
     return await Promise.all(promises);
   }
-}
 
+  analytics<T>(query: { query: IObject | IObject[] }) {
```

Comment:
Should we use the `Query` object type from the analytics module here?
If we do, and add also the `AnalyticsResponse` there would not be a need for having the `analyticsFacts.ts` service.

Reply:
I considered that, but we cannot import types from modules from the core of the application- there's a linter rule against it. This is to make sure that the core and modules stay seperate. We are already crossing that line a bit here by referring to `analytics` outside of its module, but I think that's acceptable for now

Reply:
Right, i was thinking of moving those type outside the module because they are the definition of how the analytics endpoint should be called. Maybe they belong here, and the analytics module would just be the implementation of all the charts and data handling.

Reply:
Perhaps cleaner would be to call the endpoint `query` after all? Then we don't have any reference to `analytics` in the core, and we keep them more seperate

Reply:
Now that you mention it, if we want to keep the analytics things separated we could make the method more generic by having the analytics service pass the endpoint path down to the streams method and remove any mention to analytics in there.

Reply:
Yes exactly! Shall I do it like that? What should we call the method? `query`, or something else?

Reply:
🤔 well its prettty similar to `.get` stream, but allows for a json payload, so maybe something like `.getJSON` or `.getQuery` or something like `.getSomething`?

Reply:
I'll think about it a bit